src/clr-angular/data/datagrid/datagrid-pagination.ts

```diff
@@ -3,58 +3,45 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { Component, ContentChild, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
 import { Subscription } from 'rxjs';
-
 import { Page } from './providers/page';
+import { ClrDatagridPageSize } from './datagrid-page-size';
 
 @Component({
   selector: 'clr-dg-pagination',
   template: `
-        <div class="pagination-description">
-            <ng-content></ng-content>
-        </div>
-        <ul class="pagination-list" *ngIf="page.last > 1">
-            <li *ngIf="page.current > 1">
-                <button
-                        class="pagination-previous"
-                        (click)="page.previous()"
-                        type="button"></button>
-            </li>
-            <li *ngIf="page.current > 2">
-                <button (click)="page.current = 1" type="button">1</button>
-            </li>
-            <li *ngIf="page.current > 3">...</li>
-            <li *ngFor="let pageNum of middlePages" [class.pagination-current]="pageNum === page.current">
-                <button
-                        *ngIf="pageNum !== page.current; else noButton"
-                        (click)="page.current = pageNum"
-                        type="button">{{pageNum}}
-                </button>
-                <ng-template #noButton>{{pageNum}}</ng-template>
-            </li>
-            <li *ngIf="page.current < page.last - 2">...</li>
-            <li *ngIf="page.current < page.last - 1">
-                <button
-                        (click)="page.current = page.last"
-                        type="button">{{page.last}}
-                </button>
-            </li>
-            <li *ngIf="page.current < page.last">
-                <button
-                        class="pagination-next"
-                        (click)="page.next()"
-                        type="button"></button>
-            </li>
-        </ul>
+    <div class="pagination-size" *ngIf="pageSizeComponent">
+      <ng-content select="clr-dg-page-size"></ng-content>
+    </div>
+    <div class="pagination-description">
+      <ng-content></ng-content>
+    </div>
+    <div class="pagination-list" *ngIf="page.last > 1">
+      <button class="pagination-first" [disabled]="page.current === 1" (click)="page.current = 1">
```

Comment:
The cynical engineer in me shivers at the sight of `page.current === 1` instead of `page.current <= 1`. But I'm sure it's fine, there is now way underflow would ever happen. 😬